Microsoft.DotNet.Interactive/CurrentVariables.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.Encodings.Web;
+using Microsoft.DotNet.Interactive.Formatting;
+using static Microsoft.DotNet.Interactive.Formatting.PocketViewTags;
+
+namespace Microsoft.DotNet.Interactive
+{
+    public class CurrentVariables : IEnumerable<CurrentVariable>
+    {
+        private readonly Dictionary<string, CurrentVariable> _variables = new Dictionary<string, CurrentVariable>();
+
+        public CurrentVariables(IEnumerable<CurrentVariable> variables, bool detailed)
+            : this(detailed)
+        {
+            foreach (var variable in variables)
+            {
+                _variables[variable.Name] = variable;
+            }
+        }
+
+        public CurrentVariables(Dictionary<string, CurrentVariable> variables, bool detailed)
+            : this(detailed)
+        {
+            _variables = variables ?? throw new ArgumentNullException(nameof(variables));
+        }
+
+        private CurrentVariables(bool detailed)
+        {
+            Detailed = detailed;
+        }
+
+        public bool Detailed { get; }
+
+        public IEnumerator<CurrentVariable> GetEnumerator() => _variables.Values.GetEnumerator();
+
+        IEnumerator IEnumerable.GetEnumerator() => GetEnumerator();
+
+        public static void RegisterFormatter()
```

Comment:
Rather than having this work via a side effect, it might be better to have a method that returns an instance of `ITypeFormatter<CurrentVariables>`.

Reply:
Added a separate `ITypeFormatter<CurrentValue>`.  Let me know if that's what you had in mind.

Reply:
That works. There's also `Formatter.Create` if you prefer to avoid the ceremony of creating a class.